include/ginkgo/core/matrix/csr.hpp

```diff
@@ -1042,11 +1070,47 @@ class Csr : public EnableLinOp<Csr<ValueType, IndexType>>,
         this->make_srow();
     }
 
+    /**
+     * Creates a CSR matrix from already allocated (and initialized) row
+     * pointer, column index and value arrays.
+     *
+     * @note This is the same as the previous constructor but with a default
+     *       strategy.
+     */
+    template <typename ValuesArray, typename ColIdxsArray,
+              typename RowPtrsArray>
+    Csr(std::shared_ptr<const Executor> exec, const dim<2>& size,
+        ValuesArray&& values, ColIdxsArray&& col_idxs, RowPtrsArray&& row_ptrs)
+        : Csr{exec,
+              size,
+              std::forward<ValuesArray>(values),
+              std::forward<ColIdxsArray>(col_idxs),
+              std::forward<RowPtrsArray>(row_ptrs),
+              Csr::make_default_strategy(exec)}
+    {}
+
     void apply_impl(const LinOp* b, LinOp* x) const override;
 
     void apply_impl(const LinOp* alpha, const LinOp* b, const LinOp* beta,
                     LinOp* x) const override;
 
+    // TODO: This provides some more sane settings. Please fix this!
+    static std::shared_ptr<strategy_type> make_default_strategy(
+        std::shared_ptr<const Executor> exec)
+    {
+        auto cuda_exec = std::dynamic_pointer_cast<const CudaExecutor>(exec);
+        auto hip_exec = std::dynamic_pointer_cast<const HipExecutor>(exec);
```

Comment:
I thought of not doing the dpcpp cast since it would default to classical. I guess I can add it.

I'm not sure if we should throw in the load_balance/automatic default executor. I think with this version indeed it should basically never be used anymore, so we can start depreciating it? It only depends if some user somewhere had a `std::make_shared<automatical>()`.

Reply:
depreciating is also good.